packages/firebase_database/android/src/main/java/io/flutter/plugins/firebase/database/FirebaseDatabasePlugin.java

```diff
@@ -69,28 +69,52 @@ private Query getQuery(Map<String, Object> arguments) {
     }
     if (parameters.containsKey("startAt")) {
       Object startAt = parameters.get("startAt");
-      String startAtKey = (String) parameters.get("startAtKey");
-      if (startAt instanceof Boolean) {
-        query = query.startAt((Boolean) startAt, startAtKey);
-      } else if (startAt instanceof String) {
-        query = query.startAt((String) startAt, startAtKey);
-      } else if (startAt instanceof Double) {
-        query = query.endAt((Double) startAt);
-      } else if (startAt instanceof Integer) {
-        query = query.startAt((int) startAt);
+      if (parameters.containsKey("startAtKey")) {
+        String startAtKey = (String) parameters.get("startAtKey");
+        if (startAt instanceof Boolean) {
+          query = query.startAt((Boolean) startAt, startAtKey);
+        } else if (startAt instanceof String) {
+          query = query.startAt((String) startAt, startAtKey);
+        } else if (startAt instanceof Integer) {
```

Comment:
The cases for the Integer type can be removed and ints can be processed like other numbers.  It looks like Firebase's Query only handles double/string/bool values and doesn't have any special handling for ints.